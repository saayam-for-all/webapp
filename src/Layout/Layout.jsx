import React, { Suspense } from "react";
import { Outlet } from "react-router-dom";
import Footer from "../common/components/Footer/Footer";
import LeftAds from "../common/components/LeftAds/LeftAds";
import MainLoader from "../common/components/Loader/MainLoader";
import Navbar from "../common/components/Navbar/Navbar";
import RightAds from "../common/components/RightAds/RightAds";

const Layout = () => {
  return (
    <div className="flex flex-col h-screen">
      {/* header includes Navbar which spans full width */}
      <header className="sticky z-10">
        <Navbar />
      </header>

      {/* main content */}
      <div className="flex flex-1">
        <aside className="left-ads-panel flex-1 ">
          <LeftAds />
        </aside>
        <main className="flex-[6] overflow-auto">
          <Suspense fallback={<MainLoader />}>
            <Outlet />
          </Suspense>
        </main>
        <aside className="right-ads-panel flex-1 ">
          <RightAds />
        </aside>
      </div>

      {/* footer */}
      <footer className="">
        <Footer />
      </footer>
    </div>
  );
};

export default Layout;
