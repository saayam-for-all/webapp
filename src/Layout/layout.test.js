import { render } from "@testing-library/react";
import Layout from "./Layout";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  Suspense: ({ children }) => <mock-suspense>{children}</mock-suspense>,
}));

// Mock all the custom components because they have their own snapshot tests
jest.mock("#components/Loader/MainLoader");
jest.mock("#components/Navbar/Navbar");
jest.mock("#components/Footer/Footer");
jest.mock("#components/LeftAds/LeftAds");
jest.mock("#components/RightAds/RightAds");
jest.mock("#components/NavigationGuard/NavigationGuard");

describe("Layout", () => {
  it("renders correctly", () => {
    const tree = render(<Layout />);
    expect(tree).toMatchSnapshot();
  });
});
