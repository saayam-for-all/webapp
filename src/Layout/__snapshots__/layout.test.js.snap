// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col h-screen"
      >
        <mock-navigation-guard />
        <header
          class="sticky z-10"
        >
          <mock-navbar />
        </header>
        <div
          class="flex flex-1"
        >
          <aside
            class="left-ads-panel flex-1 "
          >
            <mock-left-ads />
          </aside>
          <main
            class="flex-[6] overflow-auto"
          >
            <mock-suspense>
              <mock-outlet />
            </mock-suspense>
          </main>
          <aside
            class="right-ads-panel flex-1 "
          >
            <mock-right-ads />
          </aside>
        </div>
        <footer
          class=""
        >
          <mock-footer />
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col h-screen"
    >
      <mock-navigation-guard />
      <header
        class="sticky z-10"
      >
        <mock-navbar />
      </header>
      <div
        class="flex flex-1"
      >
        <aside
          class="left-ads-panel flex-1 "
        >
          <mock-left-ads />
        </aside>
        <main
          class="flex-[6] overflow-auto"
        >
          <mock-suspense>
            <mock-outlet />
          </mock-suspense>
        </main>
        <aside
          class="right-ads-panel flex-1 "
        >
          <mock-right-ads />
        </aside>
      </div>
      <footer
        class=""
      >
        <mock-footer />
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
